polytope_server/common/queue/sqs_queue.py

```diff
@@ -0,0 +1,105 @@
+import json
+import logging
+from . import queue
+import boto3
+import os
+
+class SQSQueue(queue.Queue):
+    def __init__(self, config):
+        
+        host = config.get("host", "localhost")
+        queue_name = config.get("name", "default")
+        region = config.get("region", "eu-central-2")
+        self.keep_alive_interval = config.get("keep_alive_interval", 60)
+        self.visibility_timeout = config.get("visibility_timeout", 120)
+        self.queue_url = "{}/{}".format(host, queue_name)
+        
+        logging.getLogger("sqs").setLevel("WARNING")
+        session = boto3.Session(
+            aws_access_key_id=os.getenv('POLYTOPE_S3_ACCESS_KEY'),
+            aws_secret_access_key=os.getenv('POLYTOPE_S3_SECRET_KEY'),
+            region_name=region
+        )
+        self.client = session.client('sqs')
+        self.check_connection()
+
+    def enqueue(self, message):
+        self.client.send_message(
+        QueueUrl=self.queue_url,
+        MessageBody=json.dumps(message.body)
+        )
+
+    def dequeue(self):
+        response = self.client.receive_message(
+            QueueUrl=self.queue_url,
+            VisibilityTimeout=self.visibility_timeout, #If processing takes more seconds, message will be read twice
+            MaxNumberOfMessages=1,
+            )
+        if not response['Messages']:
+            return None
+        if len(response['Messages']) > 1:
+            raise ValueError("Received {} messages, should have received 1".format(len(response['Messages'])))
```

Comment:
Perhaps in this case, only the first message should be processed and the others not acknowledged